src/bcs/MFEMScalarDirichletBC.C

```diff
@@ -13,8 +14,9 @@ MFEMScalarDirichletBC::validParams()
 
 MFEMScalarDirichletBC::MFEMScalarDirichletBC(const InputParameters & parameters)
   : MFEMEssentialBC(parameters),
-    _coef(
-        getMFEMProblem().makeScalarCoefficient<mfem::ConstantCoefficient>(getParam<Real>("value")))
+    _coef(getMFEMProblem().getCoefficients().declareScalar<mfem::ConstantCoefficient>(
+        "__DirichletBC" + std::to_string(reinterpret_cast<intptr_t>(this)),
```

Comment:
I believe MOOSE objects should have a unique name for the object in the `InputParameters` passed in to label the object under construction - may be more readable than converting the pointer to a string. Something like 
`parameters.get<std::string>("_unique_name")` may be preferable here?